src/components/App.js

```diff
@@ -1,14 +1,94 @@
 import React from 'react';
 import { hot } from 'react-hot-loader/root';
+import { checkURIforError, getTokenFromURI, getAuthQuery } from '../config/authConfig';
 import macaroon from '../assets/img/Macaroonicon.png';
 
-function App() {
-  return (
-    <div className="app">
-      <h1 className="title">Shmotify.</h1>
-      <img src={macaroon} alt="" />
-    </div>
-  );
+
+class App extends React.Component {
+  state = {
+    token: null,
+    user: null,
+    error: false,
+  }
+
+  componentDidMount() {
+    const error = checkURIforError();
+    const token = getTokenFromURI();
+    if (token) {
+      this.setToken(token);
+      this.getProfile(token);
+    }
+    this.setError(error);
+  }
+
+  setToken = (token) => {
+    this.setState({ token });
+  }
+
+  setError = (error) => {
+    this.setState({ error });
+  }
+
+  getProfile = (token) => {
+    const authToken = token;
+    const myHeaders = new Headers();
+
+    myHeaders.append('Authorization', `Bearer ${authToken}`);
+
+    fetch('https://api.spotify.com/v1/me', {
+      method: 'GET',
+      headers: myHeaders,
+    })
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        }
+        throw Error(`Request rejected with status ${response.status}`);
+      })
+      .then((userData) => {
+        const user = userData.id;
+        this.setState({ user });
+        this.setState({ error: false });
+      })
+      .catch(() => this.setState({ error: true }));
+  }
+
+  renderLoginLink = () => (

```

Comment:
```suggestion
  renderLoginLink()  (
```
Using an arrow function binds the function to the class, giving you access the `this.props` and `this.state` Since you're accessing neither, no need to bind this function.